servatrice/servatrice.sql

```diff
@@ -204,7 +204,7 @@ CREATE TABLE IF NOT EXISTS `cockatrice_bans` (
   KEY `time_from` (`time_from`,`ip_address`),
   KEY `ip_address` (`ip_address`),
   FOREIGN KEY(`id_admin`) REFERENCES `cockatrice_users`(`id`)  ON DELETE CASCADE ON UPDATE CASCADE
-) ENGINE=INNODB DEFAULT CHARSET=utf8;
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 DEFAULT COLLATE utf8mb4_unicode_ci;
```

Comment:
```
ERROR 1071 (42000) at line 194: Specified key was too long; max key length is 767 bytes
```

Reply:
Looks like the index on `ip_address` can't be created because 255*4 bytex exceeds the max key length of 767 bytes; this is a limit of mysql < 5.7. I'll lave a look at this soon.

Reply:
If I recall correctly this index was added to assist with slow queries during login when looking up users previous ban history.  If you wanted to keep the same collation settings there may be a way to improve that query in general and drop the key.

Reply:
The max length of an ipv6 address is 46 chars, so we could just reduce the column size in order to make it fit. Can you please check if your database contains records where the ip_address length is greater than 46?

Reply:
```
MariaDB [servatrice]> select max(length(ip_address)) from cockatrice_bans;
+-------------------------+
| max(length(ip_address)) |
+-------------------------+
|                      26 |
+-------------------------+
1 row in set (0.13 sec)
```

But by the looks of things they are current IPv4 address scheme.  There doesnt appear to be any IPv6 address's in this DB.

Reply:
Look-e there , I learned something new today.  The ```::ffff:``` is actually used on an IPv4 network to allow for IPv6 socket requests to be responded to.  Interesting.